com.ibm.streamsx.topology/opt/python/packages/streamsx/rest_primitives.py

```diff
@@ -1160,10 +1252,119 @@ def get_view_obj(_view, rc):
+
+    def start_instance(self):
+        """Start the instance for this Streaming Analytics service.
+
+        Returns:
+            dict: JSON response for the instance start operation.
+        """
+        self._delegator.start_instance()
+
+    def stop_instance(self):
+        """Stop the instance for this Streaming Analytics service.
+
+        Returns:
+            dict: JSON response for the instance start operation.
+        """
+        self._delegator.stop_instance()
+
+    def get_instance_status(self):
+        """Get the status the instance for this Streaming Analytics service.
+
+        Returns:
+            dict: JSON response for the instance status operation.
+        """
+        return self._delegator.get_instance_status()
+
+class _StreamingAnalyticsServiceV2Delegator(object):
+    """Delegator for IAM access to a Streaming Analytics service
+    """
+    def __init__(self, rest_client, credentials):
+        """
+        Args:
+            rest_client (_IAMStreamsRestClient): The client used to make REST calls.
+            credentials (str): credentials for accessing Streaming Analytics service.
+        """
+        self.rest_client = rest_client
+        self._credentials = credentials
+        self._v2_rest_url = self._credentials[IAMConstants.V2_REST_URL]
+
+    def cancel_job(self, job_id=None, job_name=None):
+        if job_id is None and job_name is None:
+            raise ValueError("Please specify either the job id or job name when cancelling a job.")
+        
+        if job_id is None:
+            # Get the job id using the job name, since it's required by the REST API
+            req = requests.Request("GET", self._v2_rest_url + '/jobs/',
+                               headers = {'Authorization' : self.rest_client._get_authorization(),
+                                          'Accept' : 'application/json'})
+            prepared = req.prepare()
+            res = self.rest_client.session.send(prepared).json()
+            self.rest_client.handle_http_errors(res)
+            for job in res['resources']:
+                if job['name'] == job_name:
+                    # Find the correct job_id, set it
+                    job_id = job['id']
+
+        # Cancel the job using the job id
+        req = requests.Request("DELETE", self._v2_rest_url + '/jobs/' + job_id,
+                               headers = {'Authorization' : self.rest_client._get_authorization(),
+                                          'Accept' : 'application/json'})
+        prepared = req.prepare()
+        res = self.rest_client.session.send(prepared)
+        self.rest_client.handle_http_errors(res)
+
+    def start_instance(self):
+        req = requests.Request("PATCH", self._v2_rest_url, json={'state' : 'STARTED'},
+                               headers = {'Authorization' : self.rest_client._get_authorization(),
+                                          'Content-Type' : 'application/json',
```

Comment:
Curious if `Content-Type` is defined already by passing in the body as `json={...}`?